### PR TITLE
The Custom Function and an Examples Folder

### PR DESCRIPTION
**Pull request by kirkw**
Pavel, 
  I am learning how to do this.  I created a file with the same name and a .md extension for the readme.  I like this approach, but not sure.  Feel free to reject this and tell me to do it correctly.

  My idea behind the examples folder is to NOT clutter up.  Also I may have used UPPER case "Examples".  
Please execute this function in a separate db (I did, but my environment could have some tool installed I accidentally used).

  My hope is to have other variations of this.  As well as to have other contributors.
I may even created a PRIZE offering for submissions of good ideas (Like 4 winners $100,$50,$25,$25 Amazon Gift Cards).
For those that submit good functions and .md files to explain them!

  Thank you, as always for your support in this process.

### CODE REVIEW

File: examples/custom_scan_function.sql [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+                check_msg := str_out || chr(10) || '######## ==> details unavailable, parsing error(2) <=== #########'::TEXT;
+                RETURN NEXT;
+                CONTINUE;
+            END IF;
+            fname := LEFT(fdecl, pos - 1); -- exclude the paren
+            fname := SUBSTR(fname, POSITION(' ' IN fname) + 1);
+            parm1 := TRIM(SUBSTR(fdecl, pos, POSITION(')' IN fdecl) - pos + 1));
+            --       RETURN NEXT (ts , concat('#### ', fdecl ));  -- Really Just Debug!
+            BEGIN
+                SELECT TO_REGPROC(fname)::oid INTO foid;  -- This function will not throw an exception, just returns NULL
```
Comment (okbob) [inline examples/custom_scan_function.sql:93]: please use :=  instead SELECT INTO

File: examples/custom_scan_function.sql [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                -- REPLACES the error block
+                IF (foid IS NULL) THEN
```
Comment (okbob) [inline examples/custom_scan_function.sql:95]: the brackets should not be used here

**Review by okbob — CHANGES_REQUESTED** [review]

**Review by kirkw — COMMENTED** [review]
It let me edit, but I am new to this.  Let me know if you see the changes the way you expect.

**Comment by okbob** [comment]
Merged, thank you for PR

**Pull request merged**